### PR TITLE
Proposed ember-moment v10

### PR DESCRIPTION
**Pull request by ef4**
The switch from ember-cli-moment-shim to ember-auto-import was a good step (ember-auto-import is more future-aligned with general ES-module-compatible build systems). But the way ember-moment currently (9.x) brings both moment and moment-timezone as dependencies and auto-imports them is problematic.

1. For the app and ember-moment to both gain access to moment and guarantee that it's the exact same copy, moment should be a peerDep of ember-moment, not a dep.
2. It's not correct to depend on both moment and moment-timezone. moment-timezone already has moment as a dependency. It's one or the other.
3. When consuming moment or moment-timezone via webpack (which is what ember-auto-import is doing), there are extra steps needed to control the locales and timezone data:
    - https://momentjs.com/docs/#/use-it/webpack/
    - https://momentjs.com/timezone/docs/#/use-it/webpack/

There are several other things I will be addressing here too:

 - Ember 4.0 compatibility
 - Providing utilities that app authors should use to take care of the aforementioned webpack settings under either ember-auto-import or embroider
 - Detecting the presence of ember-cli-moment-shim and warning app authors that it's giving them a duplicate copy of moment (I think a lot of real apps are currently shipping moment twice without realizing it!).

This PR is a breaking change that would be ember-moment 10.0. The breaking changes are:
 - apps must add moment or moment-timezone (to satisfy the new peerDep requirement)
 - apps must have ember-auto-import >= 2 (since we're doing a breaking release anyway, this is a good time to convert ember-moment to v2 format, which means consuming apps need ember-auto-import >=2).

Big picture: moment is an end-of-life technology. My recommendation to app authors is not to rush into rewriting from moment to another library like Luxon, because [Temporal](https://tc39.es/proposal-temporal/docs/index.html) is very close to happening, and if you're going to port you might as well port directly to Temporal (which will have native browser implementations).

### CODE REVIEW

**Comment by ef4** [comment]
(This was intended to be a WIP PR, not done yet.)

**Comment by ef4** [comment]
@alexlafroscia @NullVoxPopuli I'm willing to be added a maintainer here to land this work and then support it.

File: package.json [hunk 1]
```diff
@@ -9,74 +9,134 @@
...
     "@glimmer/component": "^1.0.3",
     "@glimmer/tracking": "^1.0.3",
+    "@rollup/plugin-babel": "^5.3.0",
     "babel-eslint": "^10.1.0",
-    "ember-cli": "~3.28.0",
+    "ember-auto-import": "^2.2.4",
+    "ember-cli-3.16": "npm:ember-cli@~3.16.0",
+    "ember-cli-3.20": "npm:ember-cli@~3.20.0",
+    "ember-cli-3.24": "npm:ember-cli@~3.24.0",
+    "ember-cli-3.28": "npm:ember-cli@~3.24.0",
```
Comment (NullVoxPopuli) [inline package.json:60]:
```suggestion
    "ember-cli-3.28": "npm:ember-cli@~3.28.0",
```

File: test-app/scenarios.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    baseDir: __dirname,
+    resolveName: 'ember-cli-3.28',
+  });
+  project.linkDevDependency('ember-source', {
+    baseDir: __dirname,
+    resolveName: 'ember-source-3.28',
+  });
+}
+
+// async function release(project) {
```
Comment (NullVoxPopuli) [inline test-app/scenarios.js:73]: how come these are commented out?

**Comment by NullVoxPopuli** [comment]
This is exciting work! thanks for this!

File: test-app/scenarios.js (hunk 2 above)
Comment (ef4) [reply test-app/scenarios.js:73]: That's still WIP. I intend to get those passing before this is ready.

**Comment by ef4** [comment]
This is ready to release. Can I get auth on npm please?

**Comment by NullVoxPopuli** [comment]
Done

**Pull request merged**